### PR TITLE
Add submission git handling

### LINKED ISSUE #12: Project Submission
Attendees need to be able to submit their projects. We will support project submissions as a link to a git repository.
The project "description" will be extracted from the repo's README  and cached on the server. This file must be named `README` or `README.md` or `README.txt` and must be a plain text file.

The file will be cached on the server and then loaded whenever it is requested. The content of the README will be periodically updated afterwards.

The database will need to associate the following:
* A project name
* The project git repo
* The cached project README (as a file)
* (future) The project image banner

### LINKED ISSUE #15: Sponsor Challenges
- [] CRUD endpoints for Challenges (for sponsor and organizer use)
- [] Link Sponsor Challenges to submissions

### PR DESCRIPTION
**Pull request by dellisd**

### CODE REVIEW

File: src/controllers/submissionController.ts [hunk 1]
```diff
@@ -110,24 +115,32 @@ export const submitProject = async (
...
+  if(!user) {
+      res.sendStatus(401);
+      return;
   }
 
-  // TODO: Check if user is in team, and that team has not already submitted project
+  if(!user.team) {
+    res.status(400).send('User is not on a team');
+    return;
+  }
```
Comment (dellisd) [inline src/controllers/submissionController.ts:156]: How are we handling teams that have already submitted? Are we just going with replacing the existing submission if they submit again?

**Review by dellisd — COMMENTED** [review]
Is it possible to tie the hacking end time into the submissions endpoint to cut off any new submissions after hacking ends? (perhaps with a small grace period)

File: src/controllers/submissionController.ts (hunk 1 above)
Comment (ryanlyrl) [reply src/controllers/submissionController.ts:156]: I use Object.assign, so basically, any provided field will get overwritten if they submit again. So essentially, it edits their submission.

**Comment by dellisd** [comment]
I can't approve this PR (since I authored it) but it looks good to me. As long as the git stuff is fine with you, you can approve + merge it.

**Pull request merged**